app/jobs/regular/group_smtp_email.rb

```diff
@@ -4,13 +4,48 @@
 
 module Jobs
   class GroupSmtpEmail < ::Jobs::Base
+    include Skippable
+
     sidekiq_options queue: 'critical'
 
     def execute(args)
+      return if quit_email_early?
+
       group = Group.find_by(id: args[:group_id])
       post = Post.find_by(id: args[:post_id])
       email = args[:email]
       cc_addresses = args[:cc_emails]
+      recipient_user = ::UserEmail.find_by(email: email, primary: true)&.user
+
+      if post.blank?
+        return create_skipped_email_log(
+          email_type: :group_smtp,
+          to_address: email,
+          user_id: recipient_user&.id,
+          post_id: nil,
+          reason_type: SkippedEmailLog.reason_types[:group_smtp_post_deleted]
+        )
```

Comment:
A bit of code duplication happening here which is a sign that we can extract stuff into a method.